htdocs/core/class/html.form.class.php

```diff
@@ -2305,7 +2305,7 @@ public function select_produits($selected = '', $htmlname = 'productid', $filter
 				}
 			}
 			// mode=1 means customers products
-			$urloption = 'htmlname='.$htmlname.'&outjson=1&price_level='.$price_level.'&type='.$filtertype.'&mode=1&status='.$status.'&status_purchase='.$status_purchase.'&finished='.$finished.'&hidepriceinlabel='.$hidepriceinlabel.'&warehousestatus='.$warehouseStatus;
+			$urloption = 'htmlname='.$htmlname.'&outjson=1&price_level='.$price_level.'&type='.$filtertype.'&mode=1&status='.$status.'&status_purchase='.$status_purchase.'&finished='.$finished.'&hidepriceinlabel='.$hidepriceinlabel.'&warehousestatus='.$warehouseStatus.($limit > 0 ? '&limit='.$limit : '');
```

Comment:
We are in case of PRODUIT_USE_SEARCH_TO_SELECT is on (so in case we want the limit managed by the first chars we enter and not by a setup). So we should not have parameter limit here.